src/pkg/oci/utils.go

```diff
@@ -26,34 +28,36 @@ import (
 // prepending the provided prefix
 //
 // appending the provided suffix to the version
-func ReferenceFromMetadata(registryLocation string, metadata *types.ZarfMetadata, suffix string) (*registry.Reference, error) {
-	registryLocation = strings.TrimPrefix(registryLocation, utils.OCIURLPrefix)
-
+func ReferenceFromMetadata(registryLocation string, metadata *types.ZarfMetadata, suffix string) (string, error) {
 	ver := metadata.Version
 	if len(ver) == 0 {
-		return nil, errors.New("version is required for publishing")
+		return "", errors.New("version is required for publishing")
 	}
 
 	if !strings.HasSuffix(registryLocation, "/") {
 		registryLocation = registryLocation + "/"
 	}
+	registryLocation = strings.TrimPrefix(registryLocation, utils.OCIURLPrefix)
 
 	format := "%s%s:%s-%s"
 
 	raw := fmt.Sprintf(format, registryLocation, metadata.Name, ver, suffix)
 
-	message.Debugf("raw OCI reference from metadata: %s", raw)
+	message.Debug("Raw OCI reference from metadata:", raw)
 
 	ref, err := registry.ParseReference(raw)
 	if err != nil {
-		return nil, err
+		return "", err
 	}
 
-	return &ref, nil
+	return ref.String(), nil
```

Comment:
why not return the full reference if it is this easy to get a string out?

Reply:
This function is only really used to feed into `OrasRemote`, which accepts a string parameter, and not for true full manipulation of the ref.